### PR TITLE
CNV-36798: Provisioning percentage

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

Provisioning percentage for one or multiple disks

## 🎥 Demo
image
Screenshot from 2024-03-11 18-02-49
Screenshot from 2024-03-11 17-57-59

### CODE REVIEW

File: src/utils/resources/vm/hooks/useProvisioningPercentage.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { DataVolumeModelGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import type { V1beta1DataVolume } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { getNamespace } from '@kubevirt-utils/resources/shared';
+import { getVolumes } from '@kubevirt-utils/resources/vm';
+import { useK8sWatchResources, WatchK8sResource } from '@openshift-console/dynamic-plugin-sdk';
+import { printableVMStatus } from '@virtualmachines/utils';
+
+export type NameWithPercentages = { [name in string]: string };
```
Comment (metalice) [inline src/utils/resources/vm/hooks/useProvisioningPercentage.tsx:9]: if its export not by mistake and consumed by another component, please add it to types file

File: src/utils/resources/vm/hooks/useProvisioningPercentage.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  const watchRequest = dataVolumeNames.reduce((acc, current) => {
+    acc[current] = {
+      groupVersionKind: DataVolumeModelGroupVersionKind,
+      name: current,
+      namespace: namespace,
+    };
+    return acc;
+  }, {} as { [key in string]: WatchK8sResource });
+
+  const vmPrintableStatus = vm?.status?.printableStatus;
```
Comment (metalice) [inline src/utils/resources/vm/hooks/useProvisioningPercentage.tsx:32]: all across ur pr u can use getVMStatus to get it

File: src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabDetails/components/VirtualMachineStatusWithPopover/VirtualMachineProvisioningStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  const { t } = useKubevirtTranslation();
+  const { percentages } = useProvisioningPercentage(vm);
+
+  const numberOfProvisioningDisks = Object.keys(percentages).length;
+
+  const location = useLocation();
+
+  return (
+    <>
```
Comment (metalice) [inline src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabDetails/components/VirtualMachineStatusWithPopover/VirtualMachineProvisioningStatus.tsx:32]: its needed?

Comment (upalatucci) [reply src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabDetails/components/VirtualMachineStatusWithPopover/VirtualMachineProvisioningStatus.tsx:32]: no

**Pull request merged**